mirdata/billboard.py

```diff
@@ -0,0 +1,453 @@
+                        event["notes"] = event.get("notes", []) + [i]
+                o["events"] = o.get("events", []) + [event]
+    return o
+
+
+def _timed_sections(parsed):
+    """
+        Author:
+            Brian Whitman
+            brian@echonest.com
+            https://gist.github.com/bwhitman/11453443
+
+        Given a salami parse return a list of parsed chords with timestamps & deltas
+    """
+    timed_sections = []
+    tic = 0
+    for i, e in enumerate(parsed["events"]):
+        sections = []
+        try:
+            dt = parsed["events"][i + 1]["time"] - e["time"]
+        except IndexError:
+            dt = 0
+
+        section = None
+        if e.get("notes"):
+            if len(e.get("notes")) > 1:
+                section = (e.get("notes")[0], e.get("notes")[1])
+            sections.append(section)
+
+        tic = e["time"]
+        if len(sections):
+            seconds_per_chord = dt / float(len(sections))
+            for c in sections:
+                timed_sections.append(
+                    {"time": tic, "section": c, "length": seconds_per_chord}
+                )
+                tic = tic + seconds_per_chord
+    return timed_sections
+
+
+def download(data_home=None, force_overwrite=False):
+    """Download Billboard Dataset (annotations).
+    The audio files are not provided.
+
+    Args:
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+
+        force_overwrite (bool): whether to overwrite the existing downloaded data
+
+    """
+    if data_home is None:
+        data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+    info_message = """
+        Unfortunately the audio files of the McGill-Billboard dataset are not available
+        for download. If you have the McGill-Billboard dataset, place the contents into a
+        folder called McGill-Billboard with the following structure:
+            > McGill-Billboard/
+                > annotations/ 
+                > audio/
+        and copy the McGill-Billboard folder to {}
+    """.format(
+        data_home
+    )
+
+    download_utils.downloader(
+        data_home,
+        tar_downloads=[ANNOTATIONS_REMOTE, CHORDS_REMOTE, CHORDS_MIREX13],
+        file_downloads=[INDEX_REMOTE],
+        info_message=info_message,
+        force_overwrite=force_overwrite,
+    )
+
+    annotations_dir = os.path.join(data_home, "annotations")
+    if force_overwrite:
+        if os.path.exists(annotations_dir):
+            shutil.rmtree(annotations_dir)
+        shutil.move(os.path.join(data_home, DATASET_DIR), annotations_dir)
+    else:
+        if os.path.exists(os.path.join(data_home, DATASET_DIR)):
+            if not os.path.exists(annotations_dir):
+                shutil.move(os.path.join(data_home, DATASET_DIR), annotations_dir)
+            else:
+                shutil.rmtree(os.path.join(data_home, DATASET_DIR))
```

Comment:
If I'm not mistaking this should be handled by the download function if you indicate the `annotations_dir` in the `annotations_dir=` field of your remote file (e.g. in `ANNOTATIONS_REMOTE`). @rabitt ?